lib/http/2/header/encoding_context.rb

```diff
@@ -77,10 +77,11 @@ class EncodingContext
         ["www-authenticate",            ""]
       ].each { |pair| pair.each(&:freeze).freeze }.freeze
 
-      STATIC_TABLE_BY_FIELD = STATIC_TABLE
-                              .each_with_object({})
-                              .with_index { |((field, value), hs), idx| (hs[field] ||= []) << [idx, value] }
-                              .each { |pair| pair.each(&:freeze).freeze }.freeze
+      STATIC_TABLE_BY_FIELD =
+        STATIC_TABLE
+        .each_with_object({})
+        .with_index { |((field, value), hs), idx| (hs[field.freeze] ||= []) << [idx, value.freeze].freeze }
+        .each { |pair| pair.each(&:freeze).freeze }.freeze
```

Comment:
Same thing here right? Do pair values need to be frozen or only the arrays? There are probably instances of this in other places.

Reply:
indeed, arrays must be frozen to be shareable (values already were).

Reply:
I'm essentially asking if you need `pair.each(&:freeze)` here? I would assume you don't want values frozen.

Reply:
you're right, I've simplified it now, and not try to freeze values that I know are already frozen.